data_structures/arrays/permutations.py

```diff
@@ -0,0 +1,31 @@
+from itertools import permutations
+
+
+# All permutations:
+def permute(nums: list[int]) -> list[list[int]]:
+    """
+    >>> from itertools import permutations
+    >>> numbers= [1,2,3]
+    >>> all(list(nums) in permute(numbers) for nums in permutations(numbers))
+    True
+    """
+    result = []
+    if len(nums) == 1:
+        return [nums.copy()]
+    for _ in range(len(nums)):
+        n = nums.pop(0)
```

Comment:
```suggestion
    for n in nums:
```

I think you just want to loop over each element in the array?